numpy/core/overrides.py

```diff
@@ -13,6 +13,17 @@
     int(os.environ.get('NUMPY_EXPERIMENTAL_ARRAY_FUNCTION', 1)))
 
 
+array_function_like_doc = (
+    """like : array_like
+        Reference object to allow the creation of arrays which are not
+        NumPy arrays. If an array-like passed in as ``like`` supports
+        the ``__array_function__`` protocol, the result will be defined
+        by it. In this case, it ensures the creation of an array object
+        compatible with that passed in via this argument.
```

Comment:
For those reading emails, the new version of the current version of the doc string is:

> Reference object to allow the creation of arrays which are not NumPy arrays. If an array-like passed in as ``like`` supports the ``__array_function__`` protocol, the result will be defined by it. In this case, it ensures the creation of an array object compatible with that passed in via this argument.

I suppose we still need some note on the experimental nature for the moment, but don't have a good idea right now what/how.

Reply:
A link to the NEP might be nice here.

Reply:
Yep, a clear warning like `The like= keyword is still EXPERIMENTAL, see NEP-xxxx` would be nice.